src/Gui/CommandActionPy.cpp

```diff
@@ -0,0 +1,151 @@
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 51 Franklin Street,      *
+ *   Fifth Floor, Boston, MA  02110-1301, USA                              *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "Command.h"
+#include "Action.h"
+#include "PythonWrapper.h"
+
+#include "CommandActionPy.h"
+#include "CommandPy.h"
+
+
+using namespace Gui;
+
+CommandActionPy::CommandActionPy(Py::PythonClassInstance* self, Py::Tuple& args, Py::Dict& kwds)
+    : Py::PythonClass<CommandActionPy>::PythonClass(self, args, kwds)
+{
+    const char* name;
+    if (!PyArg_ParseTuple(args.ptr(), "s", &name)) {
+        throw Py::Exception();
+    }
+
+    cmdName = name;
+    cmd = Application::Instance->commandManager().getCommandByName(name);
+}
+
+CommandActionPy::~CommandActionPy()
+{
+}
+
+Py::Object CommandActionPy::getAction()
+{
+    if (!cmd) {
+        cmd = Application::Instance->commandManager().getCommandByName(cmdName.c_str());
+    }
+
+    if (cmd) {
+        Action* action = cmd->getAction();
+        PythonWrapper wrap;
+        wrap.loadWidgetsModule();
+
+        return wrap.fromQObject(action->action());
+    }
+    else {
+        return Py::None();
+    }
+}
+
+Py::Object CommandActionPy::getCommand()
+{
+    if (!cmd) {
+        cmd = Application::Instance->commandManager().getCommandByName(cmdName.c_str());
+    }
+
+    if (cmd) {
+        auto cmdPy = new CommandPy(cmd);
+        return Py::asObject(cmdPy);
+    }
+
+    return Py::None();
+}
+PYCXX_NOARGS_METHOD_DECL(CommandActionPy, getCommand)
+
+void CommandActionPy::init_type()
+{
+    behaviors().name("Gui.CommandAction");
+    behaviors().doc("Descriptor to access the action of the commands");
+    behaviors().supportRepr();
+    behaviors().supportGetattro();
+    behaviors().supportSetattro();
+    behaviors().set_tp_descr_get(&CommandActionPy::descriptorGetter);
+    behaviors().set_tp_descr_set(&CommandActionPy::descriptorSetter);
+    PYCXX_ADD_NOARGS_METHOD(getCommand, getCommand, "Descriptor associated command");
+
+    behaviors().readyType();
+}
+
+PyObject* CommandActionPy::descriptorGetter(PyObject* self, PyObject* /*obj*/, PyObject* /*type*/)
+{
+    auto cmdAction = Py::PythonClassObject<CommandActionPy>(self).getCxxObject();
+
+    return Py::new_reference_to(cmdAction->getAction());
+}
+
+int CommandActionPy::descriptorSetter(PyObject* /*self*/, PyObject* /*obj*/, PyObject* value)
+{
+    if (value) {
+        PyErr_SetString(PyExc_AttributeError, "Can't overwrite command action");
+    }
+    else {
+        PyErr_SetString(PyExc_AttributeError, "Can't delete command action");
+    }
```

Comment:
So no matter what, we will always throw/print an error?

Reply:
My proposal is that the descriptor cannot be modified after its creation. Just access the action methods.
Note if the object is created not as a class attribute but outside, a different command can be set:
```
cmdAction = Gui.CommandAction("Part_Box")
cmdAction.name   # returns "Part_Box"
#set a new command:
cmdAction.name = "Draft_Rotate" # now the action will be the associated to Draft_Rotate command.
```